packages/block-library/src/site-title/edit.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * WordPress dependencies
+ */
+import { useEntityProp, useEntitySaving } from '@wordpress/core-data';
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { RichText } from '@wordpress/block-editor';
+
+export default function SiteTitleEdit() {
+	const [ title, setTitle ] = useEntityProp( 'root', 'site', 'title' );
+	const [ isDirty, isSaving, save ] = useEntitySaving( 'root', 'site', 'title' );
+	return (
+		<>
+			<Button
+				isPrimary
+				className="wp-block-site-title__save-button"
+				disabled={ ! isDirty || ! title }
+				isBusy={ isSaving }
+				onClick={ save }
+			>
+				{ __( 'Update' ) }
+			</Button>
+			<RichText
```

Comment:
What about adding `allowedFormats={ [] }`?
Site options typically don't support HTML, and in fact if you try, for example, to make the title bold, it would be eventually output as `<strong>Site Title</strong>`.

Reply:
There's also `<PlainText>`.

Reply:
AFAIK `<PlainText>` renders as a textarea in the editor which, among other quirks, makes it very annoying to keep it styled consistently with the front end.
On WP.com we used to use it for the Site Title and Site Description blocks, but eventually I replaced them both with a `<RichText>`, making it very much easier to handle.

Reply:
`allowedFormats={ [] }` makes sense here. This was just opened when we still had the malfunctioning `formattingControls`.

I'm adding it now.

Reply:
I feel like if it doesn't support HTML at all, PlainText is the way to go. Even if we disallow formats from a RichText, it's still a RichText and someone can tweak the HTML and pass it and it will render.

We should seek to fix PlainText issues instead IMO. How can we make it easier to style...

Reply:
>it's still a RichText and someone can tweak the HTML and pass it and it will render.

Good point.

>We should seek to fix PlainText issues instead IMO. How can we make it easier to style...

Agreed.

Reply:
Should we update that block to use PlainText?

Reply:
> I feel like if it doesn't support HTML at all, PlainText is the way to go.

Exactly. If `PlainText` is inadequate in its current state, we should fix it, because semantically it's the right component to use if you don't want HTML.

Reply:
While I can see the reasons against using `RichText`, I also disagree with using `PlainText`.
`PlainText` is pretty much a simple `textarea`.
This causes the following problems, among others:

- Site Title and Description are single line (`input` fields in wp-admin), while `textarea` is natively multiline.
This means that we'll need to add a JS hack on enter keydown (or on paste multiline content, etc.).
`RichText` already has a built-in `onSplit`/`onReplace` logic that takes care of this.

- `PlainText` uses [`TextareaAutosize`](https://github.com/buildo/react-autosize-textarea) to autoresize the `textarea` height.
But it doesn't automatically resize if late-injected CSS (such as theme editor styles) changes the font size.
`RichText` simply adds `contenteditable` to any HTML element we want (e.g. `h1` for the Site Title, `p` for the Description), delegating the resize to the element itself.

- `textarea` (and other input fields) doesn't inherit the CSS from outside like any other elements.
We would end up having to style the Site Title in the front end and in the editor in two different ways (for example, we might need to explicitly providing the font family or color to the editor's Site Title, while they would be inherited from the body on the front end).
`RichText` would use the same element and CSS class as the front end, making the style consistency efforts much easier for the themes.

Reply:
These are all good reasons to improve PlainText and not use RichText to me.

1- If we one a single line text, we can add a prop for that and use an input.
2- This sounds like a bug that should be fixed
3- That's harder, and potentially we could solve it by adding font-size: inherit and things like that to PlainText.

Worst-case scenario we could refactor PlainText to use a contenteditable but we shouldn't be  using RichText as it's mean for RichText (formatted html) and we shouldn't try to hack its purpose.

Reply:
> These are all good reasons to improve PlainText and not use RichText to me.

I can give it a try, if y'all don't mind.
If we plan to move `SiteTitle` to `PlainText`, I'll also need to keep the WPCOM version consistent.

Reply:
Please do, that's great 👍

Reply:
I gave this a try and have some doubts.

> 1- If we one a single line text, we can add a prop for that and use an input.

Though, `input` can't wrap text: if the Title is longer than the `input` width, it would end up half hidden, instead of wrapping in a new line.

As of now, I've got some good enough ways to prevent line breaks (maybe replacing them with an `insertDefaultBlock`, for example), but it still feels very fragile.

> 2- This sounds like a bug that should be fixed

This is where I'm stuck.
We don't have a proper way of listening to font size changes, and `TextareaAutosize` doesn't expose any API to force a resize anyway.

> 3- That's harder, and potentially we could solve it by adding font-size: inherit and things like that to PlainText.

This was actually easier than expected: the `PlainText`'s CSS already have everything set to `inherit` 😄 

---

> Worst-case scenario we could refactor PlainText to use a contenteditable but we shouldn't be using RichText as it's mean for RichText (formatted html) and we shouldn't try to hack its purpose.

In the end, I think we are back to this.
But the fact is that `RichText` already contains all the needed logic here, it's tried and tested, and works as expected.
Should I duplicate that much code and create a new `ContentEditableText` component, in order to remove formatted HTML support and cater to a handful of new blocks?

I'm not being dismissive, but I'm unsure how to proceed.
For example, I guess I could extract the underlying `RichText` logic into a new basic component, that's more appropriate for "plainer" texts, and then import and extend it for `RichText`.